### PR TITLE
Overhaul search filters

### LINKED ISSUE #2464: [Enhancement] Make duration search filters more like YouTube
**Is your enhancement request related to a problem? Please describe.**

Sometime during 2020, YouTube renamed the search duration filters from:

`Short (< 4 minutes)`
`Long (> 20 minutes)`

to:

`Under 4 minutes`
`Over 20 minutes`

and added the additional `4 - 20 minutes` option.

**Describe the solution you'd like**

Invidious should follow through with this naming change, and add the `4 - 20 minutes` option.

**Describe alternatives you've considered**

Don't follow suit.

**Additional context**

### PR DESCRIPTION
**Pull request by syeopite**
Cherry-picked from #2215 along with changes to resolve #2464

- Decluttered the search bar by moving filters to URL parameters
- Extracted filters box into a separate component
- Slightly improved design
- Add ability to remove filters by clicking on applied ones
- Rename duration filters (See #2464)
- Add `4 - 20 minutes` as a duration filter (See #2464)

### CODE REVIEW

File: src/invidious/views/components/channel-information.ecr [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+                          </li>
+                      <% end %>
+              <% end %>
+
+              <% if !channel.auto_generated %>
+                  <li class="pure-menu-item navbar" style="margin: 0 0 0 0.5em">
+                      <div class="searchbar">
+                          <form class="pure-form" action="/search" method="get">
+                              <fieldset>
+                                  <button class="search-button"><i class="icon ion-md-search"></i></button>
```
Comment (SamantazFox) [inline src/invidious/views/components/channel-information.ecr:100]: For acessibility, this button should have the `type=submit` HTML attribute. Also, it should be placed _after_ the text field to be in logical tab order (you can also use `tabindex`, but reordering is preferred)

File: src/invidious/views/components/channel-information.ecr [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                                  <input type="search" name="q" placeholder="<%= translate(locale, "search") %>" value="<%= env.get?("search").try {|x| HTML.escape(x.as(String)) } %>">
```
Comment (SamantazFox) [inline src/invidious/views/components/channel-information.ecr:101]: Again for acessibility: this input should have the attribute `type="text"` (`type="search"` is not a standard thing) and an `aria-label` to describe what it's for ("Channel search")

File: src/invidious/views/search.ecr [hunk 3]
```diff
@@ -1,4 +1,10 @@
+<%
+ # Satify compiler
```
Comment (SamantazFox) [inline src/invidious/views/search.ecr:2]:
```suggestion
 # Satisfy compiler
```

File: src/invidious/views/search.ecr [hunk 4]
```diff
@@ -1,4 +1,10 @@
+<%
+ # Satify compiler  
+%>
+<% filter_params = nil %>
```
Comment (SamantazFox) [inline src/invidious/views/search.ecr:4]: you can use a single `<% %>` pair for the comment and the variable. This generates less empty lines in the output.

**Review by SamantazFox — CHANGES_REQUESTED** [review]

**Comment by SamantazFox** [comment]
Also, as an afterthough, it would be better in my opinion to make the search filters a form, with an "apply" button. In addition to improving acessibility, it would make them more friendly to use (no need to refresh the page on each time a filter is added/removed/changed). And if you ask how to both have the search box and the filters as forms, just add hidden fields in both, so they'll be sent again in all cases.

File: src/invidious/views/components/channel-information.ecr [hunk 5]
```diff
@@ -0,0 +1,146 @@
+<% content_for "header" do %>
```
Comment (matthewmcgarvey) [inline src/invidious/views/components/channel-information.ecr:1]: Is this file in use somewhere? Doesn't seem to be rendered anywhere in this PR

Comment (SamantazFox) [reply src/invidious/views/components/channel-information.ecr:1]: Probably leftovers of #2215 and/or #2546

**Comment by SamantazFox** [comment]
Bump

**Comment by SamantazFox** [comment]
Closing as #2956 has been merged.
I'll work on rebasing #2546 soon!

**Pull request closed without merging**